go/vt/vtadmin/cluster/cluster.go

```diff
@@ -373,50 +374,269 @@ func (c *Cluster) getTablets(ctx context.Context) ([]*vtadminpb.Tablet, error) {
 	return c.parseTablets(rows)
 }
 
-// GetSchema returns the schema for a GetSchemaRequest on the given tablet. The
-// caller is responsible for making at least one call to c.Vtctld.Dial prior to
-// calling this function.
+// GetSchemaOptions contains the options that modify the behavior of the
+// (*Cluster).GetSchema method.
+type GetSchemaOptions struct {
+	// Tablets is the starting set of tablets that GetSchema will filter to find
+	// suitable tablet(s) to make GetSchema RPC(s) to.
+	//
+	// If empty, GetSchema will first call (*Cluster).FindTablets() to fetch all
+	// tablets for the keyspace.
+	Tablets []*vtadminpb.Tablet
+	// BaseRequest is used to share some common parameters to use for the
+	// individual tablet GetSchema RPCs made by (*Cluster).GetSchema, which
+	// takes a copy of this request in order to makeb certain overrides as
+	// needed, so these mutations are transparent to the caller.
+	//
+	// The TabletAlias field is ignored completely by (*Cluster).GetSchema, as
+	// it is overwritten for each tablet RPC that method makes.
+	//
+	// The TableSizesOnly field is overwritten only in certain tablet RPCs when
+	// SizeOpts.AggregateSizes is true. In order to move minimal bytes over the
+	// wire, we assume that schema definitions match across all shards, so we
+	// can get the full schema from just one tablet, and then just the table
+	// size information from the other N-1 tablets.
+	//
+	// The TableNamesOnly field is untouched by (*Cluster).GetSchema when not
+	// doing size aggregation. However, when doing size aggregation, if
+	// TableNamesOnly is true, we log a warning and override it. This is because
+	// TableNamesOnly is mutually exclusive with TableSizesOnly, and size
+	// aggregation requires setting TableSizesOnly in the cases described above.
+	BaseRequest *vtctldatapb.GetSchemaRequest
+	// SizeOpts control whether the (*Cluster).GetSchema method performs
+	// cross-shard table size aggregation (via the AggregateSizes field), and
+	// the behavior of that size aggregation (via the IncludeNonServingShards
```

Comment:
Going to tag @rafael just to consolidate the discussion on this thread, since it's in at least two places now.

I'm actually having trouble articulating why we would ever want this (maybe if we were mid-split and wanted to see what the true total size of provisioned storage was vs what the logical size is (because in a split you would be double-counting between a source and its destination shards), but beyond that I'm not sure).

I think, though, that if we are going to keep it, `false` is the right default here. Because it's probably confusing if you weren't explicitly opting in to the potential double-counting.

Buttttt like I just said, maybe we should remove it entirely and just always skip non-serving shards. I'm curious to hear what you both think!

Reply:
I'd be in favor of always skipping non serving shards for now. If we see an use case in the future, then we iterate and add it.

Reply:
Omitting it sounds good to me, especially since it'll simplify the front-end a lil bit too. :)